File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    answer = 0
    arr_1 = []
    while True:
        for i in arr:
            if i >= 50 and i % 2 == 0:
                arr_1.append(i//2)
            elif i <50 and i % 2 == 1:
                arr_1.append(2*i + 1)
            else:
                arr_1.append(i)
        if arr == arr_1:
            break
        else:
            arr = arr_1
            arr_1 = []
            answer += 1
    return answer

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.java
import java.util.Arrays;

class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        int[] before = new int[arr.length];
        while (true) {
            for (int i = 0; i < arr.length; i++) {
                before[i] = arr[i];
            }
            for (int i = 0; i < arr.length; i++) {
                int n = arr[i];
                if (n >= 50 && n % 2 == 0) {
                    arr[i] = n / 2;
                } else if (n < 50 && n % 2 == 1) {
                    arr[i] = n * 2 + 1;
                } else {
                    arr[i] = n;
                }
            }
            if (Arrays.equals(arr, before)) {
                break;
            }
            answer++;
        }
        return answer;
    }
}